Firestore/core/src/remote/datastore.cc

```diff
@@ -232,28 +232,26 @@ void Datastore::LookupDocumentsWithCredentials(
   GrpcStreamingReader* call = call_owning.get();
   active_calls_.push_back(std::move(call_owning));
 
-  // TODO(c++14): move into lambda.
-  call->Start([this, call, callback](
-                  const StatusOr<std::vector<grpc::ByteBuffer>>& result) {
-    LogGrpcCallFinished("BatchGetDocuments", call, result.status());
-    HandleCallStatus(result.status());
-
-    OnLookupDocumentsResponse(result, callback);
-
+  // TODO(c++14): lambda captures using move.
+  auto docs_callback =
+      [this, user_callback](const std::vector<grpc::ByteBuffer> result) {
+        user_callback(datastore_serializer_.MergeLookupResponses(result));
+      };
+
+  auto close_callback = [this, user_callback, call](const util::Status& status,
+                                                    bool callback_fired) {
+    // Trigger user_callback with an error status
```

Comment:
If we look up 0 documents, it will still call `docs_callback_({})` from grpc_streaming_reader.cc line 88. So this block will only be triggered with an error status.

Reply:
Got it. I missed that subtlety.